File: bialydunajec-angular-frontend/projects/bialydunajec-admin/src/app/academic-ministry/service/rest/request/create-academic-ministry.request.ts
import {PlaceDto} from '../../../../shared/service/rest/dto/place.dto';
import {ExtendedDescriptionDto} from '../../../../shared/service/rest/dto/extended-description.dto';
import {SocialMediaDto} from '../../../../shared/service/rest/dto/social-media.dto';

export class CreateAcademicMinistryRequest {
  officialName: string;
  shortName?: string;
  logoImageUrl?: string;
  place?: PlaceDto;
  socialMedia?: SocialMediaDto;
  emailAddress?: string;
  photoUrl?: string;
  description?: ExtendedDescriptionDto;

  constructor(officialName: string, shortName: string, logoImageUrl: string, place: PlaceDto, socialMedia: SocialMediaDto, emailAddress: string, photoUrl: string, description: ExtendedDescriptionDto) {
    this.officialName = officialName;
    this.shortName = shortName;
    this.logoImageUrl = logoImageUrl;
    this.place = place;
    this.socialMedia = socialMedia;
    this.emailAddress = emailAddress;
    this.photoUrl = photoUrl;
    this.description = description;
  }
}

export class UpdateAcademicMinistryRequest {
  officialName: string;
  shortName?: string;
  logoImageUrl?: string;
  place?: PlaceDto;
  socialMedia?: SocialMediaDto;
  emailAddress?: string;
  photoUrl?: string;
  description?: ExtendedDescriptionDto;

  constructor(officialName: string, shortName: string, logoImageUrl: string, place: PlaceDto, socialMedia: SocialMediaDto, emailAddress: string, photoUrl: string, description: ExtendedDescriptionDto) {
    this.officialName = officialName;
    this.shortName = shortName;
    this.logoImageUrl = logoImageUrl;
    this.place = place;
    this.socialMedia = socialMedia;
    this.emailAddress = emailAddress;
    this.photoUrl = photoUrl;
    this.description = description;
  }
}

File: backend/bialydunajec-academic-ministry/presentation/src/main/kotlin/org/bialydunajec/academicministry/rest/v1/admin/request/CreateAcademicMinistryRequest.kt
package org.bialydunajec.academicministry.rest.v1.admin.request

import org.bialydunajec.ddd.dto.sharedkernel.ExtendedDescriptionDto
import org.bialydunajec.ddd.dto.sharedkernel.PlaceDto
import org.bialydunajec.ddd.dto.sharedkernel.SocialMediaDto
import javax.validation.constraints.NotBlank

internal data class CreateAcademicMinistryRequest (
        @NotBlank
        val officialName: String,
        val shortName: String?,
        val logoImageUrl: String?,
        val place: PlaceDto?,
        val socialMedia: SocialMediaDto,
        val emailAddress: String?,
        val photoUrl: String?,
        val description: ExtendedDescriptionDto?
)
